### PR TITLE
Add tests workflow

### PR DESCRIPTION
**Pull request by kerenr-jfrog**
- [x] All [tests](https://github.com/jfrog/jfrog-visual-studio-extension/actions/workflows/tests.yml) passed. If this feature is not already covered by the tests, I added new tests.
- [x] Open pull request for dev branch.
-----

- Added a workflow for running tests when push or pull-request events occur on this repository.
- A basic structure of a test project was added for future tests implementation.

### CODE REVIEW

**Comment by kerenr-jfrog** [comment]
I have read the CLA Document and I hereby sign the CLA

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          nuget-${{ runner.os }}-
+          
+    - name: Restore dependencies
+      run: |
+        dotnet restore
+        nuget restore
+    
+    - name: Download JFrog CLI executable
+      run: |
+        powershell -Command "Start-Process -Wait -Verb RunAs powershell '-NoProfile iwr https://releases.jfrog.io/artifactory/jfrog-cli/v2-jf/${{ env.JFROG_CLI_VERSION }}/jfrog-cli-windows-amd64/jf.exe -OutFile ./JFrogVSExtension/Resources/jfrog.exe'"
```
Comment (attiasas) [inline .github/workflows/tests.yml:45]: Please use this action:
* https://github.com/jfrog/setup-jfrog-cli

Comment (attiasas) [inline .github/workflows/tests.yml:45]: Please use this action: https://github.com/jfrog/setup-jfrog-cli

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    - name: Build VSIX Project
+    # build the vsix project using multi-core compilation and parallel builds
+      run: msbuild ./JFrogVSExtension/ /p:Configuration=Release /p:Platform="AnyCPU" /p:BuildInParallel=true /m
```
Comment (attiasas) [inline .github/workflows/tests.yml:49]: Why do we have to build the project for the tests? is `restore` not enough?

File: UnitTestJfrogVSExtension/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+[assembly: AssemblyProduct("UnitTestJfrogVSExtension")]
+[assembly: AssemblyCopyright("Copyright ©  2024")]
+[assembly: AssemblyTrademark("")]
+[assembly: AssemblyCulture("")]
+
+[assembly: ComVisible(false)]
+
+[assembly: Guid("db770b26-1629-45d0-8526-8902f83731b2")]
+
+// [assembly: AssemblyVersion("1.0.*")]
```
Comment (attiasas) [inline UnitTestJfrogVSExtension/Properties/AssemblyInfo.cs:18]: delete comments

File: UnitTestJfrogVSExtension/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
+using System.Reflection;
```
Comment (attiasas) [inline UnitTestJfrogVSExtension/Properties/AssemblyInfo.cs:1]: Is this file autogenerated? why do we need `AssemblyInfo.cs` for the tests?

**Review by attiasas — CHANGES_REQUESTED** [review]
Nice!, check out my comments

File: .github/workflows/tests.yml (hunk 2 above)
Comment (kerenr-jfrog) [reply .github/workflows/tests.yml:49]: It is a part of running the test procedure so we must build the project in order to run the tests.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (kerenr-jfrog) [reply .github/workflows/tests.yml]: Using powershell script for downloading the appropriate jfrog-cli file as we agreed.

File: JFrogVSExtension/Resources/DownloadJfrogCli.ps1 [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+# Define the destination path for the downloaded file
+$destinationPath = Join-Path (Get-Location).Path "JFrogVSExtension/Resources/jfrog.exe"
+
+# Download the JFrog CLI executable 
+Invoke-WebRequest -Uri $jfrogCliUrl -OutFile $destinationPath -Verbose
+
+
+# Verify the file was downloaded successfully 
+if (Test-Path -Path $destinationPath) {
```
Comment (hadarshjfrog) [inline JFrogVSExtension/Resources/DownloadJfrogCli.ps1:18]: add a "$destinationPath --version" to validate it where 

also - compare it to `$env:JFROG_CLI_VERSION`

File: JFrogVSExtension/Resources/DownloadJfrogCli.ps1 [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# Check if the JFROG_CLI_VERSION environment variable is set, if not using latest version
+if (-not $env:JFROG_CLI_VERSION) {
```
Comment (hadarshjfrog) [inline JFrogVSExtension/Resources/DownloadJfrogCli.ps1:2]: I'd remove it - so we'll have a version lock in...

File: JFrogVSExtension.sln [hunk 7]
```diff
@@ -5,6 +5,8 @@ VisualStudioVersion = 17.1.32407.343
 MinimumVisualStudioVersion = 10.0.40219.1
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "JFrogVSExtension", "JFrogVSExtension\JFrogVSExtension.csproj", "{6443B797-2478-4A1D-BECA-28E24C1F1F41}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "UnitTestJfrogVSExtension", "UnitTestJfrogVSExtension\UnitTestJfrogVSExtension.csproj", "{DB770B26-1629-45D0-8526-8902F83731B2}"
```
Comment (hadarshjfrog) [inline JFrogVSExtension.sln:8]: Please check if can be a CONST somewhere.
Also - see how can we separate the tests from the solution

Comment (kerenr-jfrog) [reply JFrogVSExtension.sln:8]: It is possible to define in the build command which project we want to build so its ok

**Pull request merged**